### PR TITLE
[24.0] Fix bugs relating to history keyboard navigation

### PR DESCRIPTION
**Pull request by ahmedhamidawan**
Adds cross platform Ctrl/Cmd(Metakey) compatibility for features added in https://github.com/galaxyproject/galaxy/pull/17502

**Also adds a minor Shift+Click select feature (on top of the existing Shift+Arrowkey feature)**:

https://github.com/galaxyproject/galaxy/assets/78516064/b18f50ec-659f-431f-ad7b-b8d4b4fa56df

Here, the user first selects individual items using `Ctrl+Click`, and selects batches of items using `Shift+Ctrl+Click`, to be able to select batch items while maintaining prior selections.
Then, they select items using only` Shift+Click`, which restricts the selection to the initial item and the range.

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [x] Instructions for manual testing are as follows:
  1. Select an initial item using Ctrl/Shift + Click
  2. Then `Shift+Click` another item above or below the initial one to select the ones in the range
  3. Also, try out `Shift+Ctrl+Click` to select items in multiple batches

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: client/src/components/History/Content/ContentItem.vue [hunk 1]
```diff
@@ -230,6 +230,10 @@ export default {
         },
     },
     methods: {
+        isCtrlKey(event) {
+            const isMac = navigator.userAgent.indexOf("Mac") >= 0;
```
Comment (martenson) [inline client/src/components/History/Content/ContentItem.vue]: likely the best we can do, but also 🫣

File: client/src/components/History/Content/ContentItem.vue [hunk 2]
```diff
@@ -230,6 +230,10 @@ export default {
         },
     },
     methods: {
+        isCtrlKey(event) {
```
Comment (martenson) [inline client/src/components/History/Content/ContentItem.vue]:
```suggestion
        isSelectKey(event) {
```

Comment (martenson) [reply client/src/components/History/Content/ContentItem.vue]: Maybe something like this to clarify what the function does?

**Review by martenson — APPROVED** [review]
It works well, thanks @ahmedhamidawan 

As a fairly needed followup to this feature I think we should take care to enable click-to-select on a larger area of the history items when tags are present. See below what portion of the item is clickable, the rest is not and it feels weird.

Screenshot 2024-03-04 at 5 14 43 PM

File: client/src/components/History/Content/ContentItem.vue (hunk 1 above)
Comment (martenson) [reply client/src/components/History/Content/ContentItem.vue]: Maybe worth considering refactoring it to somewhere higher up in the hierarchy, since we will likely need this in other shortcut-enabled interfaces.

**Comment by ahmedhamidawan** [comment]
Yes, very good point, doing this as part of the same PR

**Comment by dannon** [comment]
Haven't done any tinkering to actually try it, but it might be worth looking into https://vueuse.org/core/useMagicKeys/#usemagickeys w/ https://vueuse.org/core/useKeyModifier/#usekeymodifier to simplify this in a standard non-platform-specific way.  And we already use vueuse.

File: client/src/components/History/Content/ContentItem.vue (hunk 2 above)
Comment (ahmedhamidawan) [reply client/src/components/History/Content/ContentItem.vue]: I was thinking more along the lines of whether the key associated to the event is Ctrl (windows) or Cmd (Mac). But, isSelectKey makes sense too

File: client/src/components/History/Content/ContentOptions.vue [hunk 3]
```diff
@@ -12,7 +12,6 @@ const props = defineProps({
     isVisible: { type: Boolean, default: true },
     state: { type: String, default: "" },
     itemUrls: { type: Object, required: true },
-    keyboardSelectable: { type: Boolean, default: true },
```
Comment (ElectronicBlueberry) [inline client/src/components/History/Content/ContentOptions.vue:15]: Why was this removed?

Comment (ahmedhamidawan) [reply client/src/components/History/Content/ContentOptions.vue:15]: The ContentOptions buttons are always Keyboard selectable now, with the Tab key. To go from one item to another, you can use the arrow keys, the tab key will take you from ContentItem -> ContentOptions -> ContentItem ....

File: client/src/components/History/Content/ContentItem.vue (hunk 2 above)
Comment (martenson) [reply client/src/components/History/Content/ContentItem.vue]: totally your call, I was just noting it reads weird when I press Command Key and the function isCtrlKey is `true`

Comment (ahmedhamidawan) [reply client/src/components/History/Content/ContentItem.vue]: Oh now I'm trying to switch over to what Dannon recommended with the magicKeys stuff

**Comment by martenson** [comment]
@dannon Likely a decent standard for the key detection. However the logic here goes beyond different keyboards, because the standard macos behavior for selecting is command+click despite having the ctrl key. I did not see that being handled in the lib you linked. Did I miss it?

**Comment by dannon** [comment]
Good point -- might not be worth caring about OS, and just have a simpler `modifierPressed` be the union of ctrl/cmd/meta/whatever?

Is there a situation where that would actually be problematic?  Would it just be something like I guess on a windows machine you get win+c copying as well, which is fine?

File: client/src/components/History/Content/ContentItem.vue (hunk 2 above)
Comment (ahmedhamidawan) [reply client/src/components/History/Content/ContentItem.vue]: But your point actually still stands now I realize,yes renaming this function will be better

**Comment by ahmedhamidawan** [comment]
> Good point -- might not be worth caring about OS, and just have a simpler `modifierPressed` be the union of ctrl/cmd/meta/whatever?
> 
> 
> 
> Is there a situation where that would actually be problematic?  Would it just be something like I guess on a windows machine you get win+c copying as well, which is fine?

Only thing is, you wouldn't want to prevent default behavior for the Windows key 

For Mac, it'd probably work out fine for Control+Click or Cmd-Click doing the same thing. It would feel weird if on Windows, Ctrl-Click and Windows-Click were the same though

**Comment by dannon** [comment]
Sure -- does windows+click have a predefined behavior?  Ctrl+click opens in a new tab, so we're already kind of overwriting that I guess.

**Comment by ahmedhamidawan** [comment]
> Sure -- does windows+click have a predefined behavior?  Ctrl+click opens in a new tab, so we're already kind of overwriting that I guess.

This would only be the behavior when you ctrl click on the area which typically expands the dataset/ opens the collection, but I see your point. Maybe then that's what I'll do here, treat the Meta and Ctrl/Control key the same way (for ContentItems)

**Comment by ahmedhamidawan** [comment]
So maybe then
```
function isSelectKey(event: KeyboardEvent) {
    const isMac = navigator.userAgent.indexOf("Mac") >= 0;
    return isMac ? event.metaKey : event.ctrlKey;
}
```
is an alright way of using the appropriate key for this? Since, for the purposes here, Ctrl-Click or Ctrl-A for Windows, and Cmd-Click or Cmd-A for Mac are default behaviors that users are used to across apps?

File: client/src/components/History/Content/ContentItem.vue (hunk 1 above)
Comment (ahmedhamidawan) [reply client/src/components/History/Content/ContentItem.vue]: ~~Would it be useful to put this in the `eventStore` and then globally available?~~ I ended up adding it there

**Comment by dannon** [comment]
@ahmedhamidawan Yeah, I think that's fine.  I didn't realize control+click on mac was also a right click; we shouldn't mask that.
